dbt/include/global_project/macros/materializations/table/table.sql

```diff
@@ -63,20 +63,34 @@
     {%- endif -%}
   {%- endcall %}
 
-  {{ run_hooks(post_hooks, inside_transaction=True) }}
-
   -- cleanup
   {% if non_destructive_mode -%}
     -- noop
   {%- else -%}
     {% if old_relation is not none %}
       -- move the existing relation out of the way
-      {{ adapter.rename_relation(target_relation, backup_relation) }}
+      {# /*
+         If this is a _view_ on _snowflake_ AND the contract of the _view_ was broken
+         by a change to a parent model, then this can throw when `alter table rename` runs, eg:
+               View definition for <relation> declared 1 column(s), but view query produces 2 column(s)
+
+         Instead, drop this view to avoid this potential error. Note: This won't work on Redshift,
+         as dropping inside the transaction will lead to a "table dropped by concurrent transaction"
+         error. In the future, Snowflake should use `create or replace table` syntax to obviate this code
+      */ #}
+      {% if adapter.type() == 'snowflake' and old_relation.type == 'view' %}
```

Comment:
🤢 

This is antithetical to the adapter architecture. Why not implement totally different table and view materializations for Snowflake vs. Bigquery? Did we architect this part incorrectly?